lib/ansible/modules/cloud/vmware/vmware_resource_pool.py

```diff
@@ -275,16 +264,21 @@ def state_add_rp(self):
         rp_spec.memoryAllocation = mem_alloc
 
         self.dc_obj = find_datacenter_by_name(self.content, self.datacenter)
-        self.cluster_obj = find_cluster_by_name_datacenter(
-            self.dc_obj, self.cluster)
+        if self.dc_obj is None:
+            self.module.fail_json(msg="Unable to find datacenter with name %s" % self.datacenter)
+
+        self.cluster_obj = find_cluster_by_name(self.content, self.cluster)
```

Comment:
Shouldn't this be `find_cluster_by_name(self.content, self.cluster, dc_obj)`? As you have the DC, it will speed up the search